### ISSUE TITLE
Problem with cargo clippy --fix

### ISSUE THREAD

**Issue by holtgrewe** [rust-lang/rust-clippy#10718]

**Comment by blyxyas** [comment]
I don't think this problem can be solved without almost re-writing a big part of the original lint. I'll unnassign myself and maybe try in a few months, if someone else wants to try, it's free.

@rustbot label: +E-hard

**Comment by Alexendoo** [comment]
Creating such a suggestion would be out of scope of clippy since it would approach a global analysis, but for the issue at hand changing the applicability to `MaybeIncorrect` would be appropriate so that `--fix` doesn't apply it automatically
